etc/cmake/options.cmake

```diff
@@ -118,6 +118,7 @@ ot_option(OT_NAT64_BORDER_ROUTING OPENTHREAD_CONFIG_NAT64_BORDER_ROUTING_ENABLE
 ot_option(OT_NAT64_TRANSLATOR OPENTHREAD_CONFIG_NAT64_TRANSLATOR_ENABLE "NAT64 translator support")
 ot_option(OT_NEIGHBOR_DISCOVERY_AGENT OPENTHREAD_CONFIG_NEIGHBOR_DISCOVERY_AGENT_ENABLE "neighbor discovery agent")
 ot_option(OT_NETDATA_PUBLISHER OPENTHREAD_CONFIG_NETDATA_PUBLISHER_ENABLE "Network Data publisher")
+ot_option(OT_NETWORK_NAME_ZERO_LENGTH OPENTHREAD_CONFIG_NETWORK_NAME_ZERO_LENGTH_ENABLE "enable zero length network name support")
```

Comment:
May be ok to not add this in `options.cmake` and have it only be a header config.

Generally the CMake option are used to enable/disable full/bigger features (then we may have a bunch of other configs per features to change parameters or behaviors).

Reply:
Hey thanks for these comments, this is helpful background and I'll make the updates.

I see there's a few tests failing that don't really seem related to what I've changed here. Do you or Jonathan have any suggestions there? Maybe just because I haven't rebased onto main in a few days?

Reply:
I kicked the failed test to re-run (sometimes it is unrelated to the change in PR and just GH action or test being flaky). 
If they keep failing after multiple runs, then we may need to look into to see why.